plugins-compat-tester/src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```
@@ -649,6 +646,74 @@ public void cloneFromSCM(PomData pomData, String name, String version, File chec
+        // 'git' 'clone' '--depth' '1' '--branch' 'scmTag' 'git url'
+        // 'git' 'ls-remote' 'giturl'
+        // 'git' 'fetch' 'giturl'
+        // 'git' 'checkout' 'scmTag'
+        // 'git' 'ls-files'
+
+        // we try something quick as the maven scm is doing clone and another useless fetch
+        // git init
+        // git fetch --depth=1 url scmTag (this will work with SHA1 or tag)
+        // git checkout FETCH_HEAD
+
+        if (Files.exists(checkoutDirectory.toPath())) {
+            org.apache.commons.io.FileUtils.deleteDirectory(checkoutDirectory);
+        }
+        if (!checkoutDirectory.mkdirs()) {
+            System.out.println("cannot create directory " + checkoutDirectory);
+            return false;
+        }
+
+        // TODO timeout??
+        int res = new ProcessBuilder().directory(checkoutDirectory).command("git", "init").inheritIO().start().waitFor();
+        if (res != 0) {
+            System.out.println("git init failed");
+            return false;
+        }
+        String gitUrl = connectionURL;
+        if (StringUtils.startsWith(connectionURL, "scm:git:")) {
+            gitUrl = StringUtils.substringAfter(connectionURL, "scm:git:");
+        }
+        List<String> commands = new ArrayList<>();
+        commands.add("git");
+        commands.add("fetch");
+        commands.add("--depth=1");
+
+        commands.add(gitUrl);
+        commands.add(scmTag);
+        res = new ProcessBuilder().directory(checkoutDirectory).command(commands).inheritIO().start().waitFor();
+        if (res != 0) {
+            System.out.println("git fetch origin failed");
+            return false;
+        }
+        res = new ProcessBuilder().directory(checkoutDirectory).command("git", "checkout", "FETCH_HEAD").inheritIO().start().waitFor();
+        if (res != 0) {
+            System.out.println("git checkout FETCH_HEAD failed");
+            return false;
+        }
+        return true;
```

Rather than returning `boolean` just throw `IOException` in case of any failure.